File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L6/06/arraylebihdarisatu.kt
package IR

import java.util.*

object arraylebihdarisatu {
    @JvmStatic
    fun main(args: Array<String>) {
        print("Enter a 4 by 4 matrix row by row: ")





        print("Sum of the elements in the major  diagonal is  " + HitungDiagonal()) //menampilkan fungsi hasil perhitungan diagonal
    }

    fun HitungDiagonal(): Double { //fungsi untuk mengitung diagonal
        val sc = Scanner(System.`in`)
        var hasil = 0.0
        val matrix = Array(4) { DoubleArray(4) } //inisialisai array 2 dimensi
        for (x in 0..3) {
            for (y in 0..3) {
                matrix[x][y] = sc.nextDouble() //menginput angka ke dalam array 2d
                if (x == y) {
                    hasil += matrix[x][y]
                }
            }
        }
        return hasil
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L6/06/arraylebihdarisatu.ts
import { Scanner } from 'readline-sync';

function HitungDiagonal(): number {
  const sc = new Scanner();
  let hasil = 0;
  const matrix: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));

  console.log('Enter a 4 by 4 matrix row by row:');

  for (let x = 0; x < 4; x++) {
    for (let y = 0; y < 4; y++) {
      matrix[x][y] = sc.nextDouble();
      if (x === y) {
        hasil += matrix[x][y];
      }
    }
  }

  return hasil;
}

console.log(`Sum of the elements in the major diagonal is ${HitungDiagonal()}`);

